client/my-sites/plans/controller.jsx

```diff
@@ -52,6 +55,12 @@ export default {
 			return;
 		}
 
+		//Redirect for upgraded sites
+		if ( site && ( isPremium( site.plan ) || isBusiness( site.plan ) ) ) {
+			page.redirect( '/plans/my-plan/' + site.slug );
```

Comment:
I think instead of redirecting we should just update the sidebar link for the "plans" item.